### PR TITLE
feat: add explicit port routing & go tools bump

### PR DESCRIPTION
**Pull request by nandor-magyar**
This contains (will contain):
- [x] go toolchain upgrades, fixes & new image
- [x] metrics port fix (use internal ports, external makes zero sense for crane)
- [x] deploymentStrategy rename and usage - it will actually be used
- [x] explicit option to select target ports when defining routing (both agent & crane)

### CODE REVIEW

**Review by polaroi8d — APPROVED** [review]
LMGT (informal)

File: web/crux/proto/common.proto [hunk 1]
```diff
@@ -137,6 +137,7 @@ message Routing {
   optional string path = 101;
   optional bool stripPath = 102;
   optional string uploadLimit = 103;
+  optional uint32 port = 104;
```
Comment (m8vago) [inline web/crux/proto/common.proto:140]: why not uint16?

**Review by m8vago — APPROVED** [review]

**Comment by m8vago** [comment]
Tested, seems working.

File: web/crux/proto/common.proto (hunk 1 above)
Comment (nandor-magyar) [reply web/crux/proto/common.proto:140]: Not available in proto, or it was bugging.

**Pull request merged**